### PR TITLE
[codex] allow CCA image generation and web search extensions

### PR DESCRIPTION
**Pull request by won-openai**
## Summary

- allow the standalone image-generation and web-search extensions for the actor-authorized provider shape used by CCA
- preserve builtin `image_generation` and `web_search` for older models and existing flows
- keep ordinary non-OpenAI providers excluded from both extensions
- remove only the image extension local managed-AuthManager requirement that CCA cannot satisfy
- share actor-authorization detection through `ModelProviderInfo`
- keep Core tests focused on routing behavior and cover header-shape edge cases in `model-provider-info`
- add a Responses Lite regression that verifies both `image_gen.imagegen` and `web.run`

## Why

CCA uses a provider named `local` with `requires_openai_auth: false` and a non-empty `x-openai-actor-authorization` header. Core accepts that provider shape, but both extension provider-name gates rejected it; image generation additionally required a Codex-managed login.

The standalone paths must coexist with existing builtin tools. New Responses Lite models can receive `image_gen.imagegen` and `web.run`, while older models continue using builtin tools.

## Impact

This enables both standalone extensions for CCA once installed downstream, without removing or changing builtin-tool compatibility for older models.

## Validation

- `just test -p codex-core responses_lite_exposes_standalone_tools_for_actor_authorized_provider`
- `just test -p codex-core responses_lite_uses_standalone_web_search_and_image_generation`
- `just test -p codex-core hosted_tools_follow_provider_auth_model_and_config_gates`
- `just test -p codex-image-generation-extension`
- `just test -p codex-web-search-extension`
- `just test -p codex-model-provider-info`
- `just fmt`
- `git diff --check`

### CODE REVIEW

**Comment by won-openai** [comment]
@codex review

File: codex-rs/ext/image-generation/src/extension.rs [hunk 1]
```diff
@@ -34,19 +34,19 @@ struct ImageGenerationExtensionConfig {
...
 
 impl ImageGenerationExtensionConfig {
-    /// Resolves whether standalone image generation should be available for a thread.
+    /// Resolves the image provider and save root for a thread.
     fn from_config(config: &Config, resolve_save_root: &SaveRootResolver) -> Self {
         Self {
-            // Core selects this executor per turn using the feature flag or model metadata.
-            available: config.model_provider.is_openai(),
+            available: config.model_provider.is_openai()
+                || config.model_provider.uses_openai_actor_authorization(),
```
[the remark this answers is not in the document]
Comment (sayan-oai) [reply codex-rs/ext/image-generation/src/extension.rs:41]: this true?

**Review by sayan-oai — APPROVED** [review]
